src/app/consensus_test/consensus_test.ml

```diff
@@ -0,0 +1,229 @@
+open Core_kernel
+open Async
+
+module BlockFileOutput = struct
+  type t =
+    { height : int; parent_state_hash : string; previous_state_hash : string }
+  [@@deriving to_yojson]
+end
+
+type select_outcome = CandidateLonger | EqualLength | CandidateShorter
+
+module type CONTEXT = sig
+  val logger : Logger.t
+
+  val constraint_constants : Genesis_constants.Constraint_constants.t
+
+  val consensus_constants : Consensus.Constants.t
+end
+
+let context (logger : Logger.t) : (module CONTEXT) =
+  ( module struct
+    let logger = logger
+
+    let proof_level = Genesis_constants.Proof_level.None
+
+    let precomputed_values =
+      { (Lazy.force Precomputed_values.for_unit_tests) with proof_level }
+
+    let consensus_constants = precomputed_values.consensus_constants
+
+    let constraint_constants = precomputed_values.constraint_constants
+  end )
+
+let read_directory dir_name =
+  let extract_height_from_filename fname =
+    let prefix_start = String.index_exn fname '-' + 1 in
+    let suffix_start = String.index_from_exn fname prefix_start '-' in
+    String.sub fname ~pos:prefix_start ~len:(suffix_start - prefix_start)
+    |> int_of_string
+  in
+  let blocks_in_dir dir =
+    let%map blocks_array = Async.Sys.readdir dir in
+    Array.sort blocks_array ~compare:(fun a b ->
+        Int.compare
+          (extract_height_from_filename a)
+          (extract_height_from_filename b) ) ;
+    let blocks_array =
+      Array.map ~f:(fun fname -> Filename.concat dir fname) blocks_array
+    in
+    Array.to_list blocks_array
+  in
+  blocks_in_dir dir_name
+
+let read_block_file blocks_filename =
+  let read_block_line line =
+    match Yojson.Safe.from_string line |> Mina_block.Precomputed.of_yojson with
+    | Ok block ->
+        block
+    | Error err ->
+        failwithf "Could not read block: %s" err ()
+  in
+  let blocks_file = In_channel.create blocks_filename in
+  match In_channel.input_line blocks_file with
+  | Some line ->
+      In_channel.close blocks_file ;
+      return (read_block_line line)
+  | None ->
+      In_channel.close blocks_file ;
+      failwithf "File %s is empty" blocks_filename ()
+
+let precomputed_block_to_block_file_output (block : Mina_block.Precomputed.t) :
+    BlockFileOutput.t =
+  let open Yojson.Safe.Util in
+  let block_json = Mina_block.Precomputed.to_yojson block in
+
+  (* Extract desired fields *)
+  let data = block_json |> member "data" in
+  let protocol_state = data |> member "protocol_state" in
+  let body = protocol_state |> member "body" in
+  let consensus_state = body |> member "consensus_state" in
+  let height =
+    consensus_state |> member "blockchain_length" |> to_string |> int_of_string
+  in
+  let parent_state_hash =
+    protocol_state |> member "previous_state_hash" |> to_string
+  in
+
+  { BlockFileOutput.height
```

Comment:
NIT: You don't need to write the module name before every field in the record, just the first one. In fact, in this case, you don't need to write it before any of the fields because you explicitly annotated the return value of the function, so OCaml will be able to automatically open the fields from the module that contains the type.